pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,291 @@
+	var h *helper.H
+	var client *resources.Resources
+	var clusterNamespace string
+	var testNodePoolName string
+	var initialNodeCount int
+
+	nodePoolGVR := schema.GroupVersionResource{
+		Group:    "hypershift.openshift.io",
+		Version:  "v1beta1",
+		Resource: "nodepools",
+	}
+
+	ginkgo.BeforeAll(func() {
+		if !viper.GetBool(config.Hypershift) {
+			ginkgo.Skip("NodePool tests are only supported on HyperShift clusters")
+		}
+
+		h = helper.New()
+		client = h.AsUser("")
+
+		var nodeList corev1.NodeList
+		expect.NoError(client.List(context.Background(), &nodeList))
+
+		if len(nodeList.Items) == 0 {
+			ginkgo.Skip("No nodes found - cannot run NodePool tests")
+		}
+
+		initialNodeCount = len(nodeList.Items)
+
+		for _, node := range nodeList.Items {
+			if label, exists := node.Labels["hypershift.openshift.io/nodePool"]; exists {
+				parts := strings.Split(label, "-workers-")
+				if len(parts) >= 1 {
+					clusterNamespace = parts[0]
+					break
+				}
+			}
+		}
+
+		if clusterNamespace == "" {
+			ginkgo.Skip("Could not determine cluster namespace from node labels")
+		}
+
+		testNodePoolName = fmt.Sprintf("test-%d", time.Now().Unix()%100000)
+	})
+
+	ginkgo.AfterAll(func(ctx context.Context) {
+		if testNodePoolName != "" {
+			ginkgo.By("Cleaning up test NodePool")
+			err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+				Delete(ctx, testNodePoolName, metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				ginkgo.GinkgoLogr.Error(err, "Failed to cleanup test NodePool", "name", testNodePoolName)
+			}
+
+			ginkgo.By("Cleaning up test pods")
+			podList := &corev1.PodList{}
+			err = client.WithNamespace(h.CurrentProject()).List(ctx, podList)
+			if err == nil {
+				for _, pod := range podList.Items {
+					if strings.HasPrefix(pod.Name, "nodepool-test-") {
+						client.Delete(ctx, &pod)
+					}
+				}
+			}
+		}
+	})
+
+	ginkgo.It("should successfully create NodePool", func(ctx context.Context) {
+		ginkgo.By("Getting existing NodePool configuration")
+
+		existingNodePools, err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).List(ctx, metav1.ListOptions{})
+		expect.NoError(err, "Failed to list existing NodePools")
+		Expect(len(existingNodePools.Items)).To(BeNumerically(">", 0), "No existing NodePools found to reference")
+
+		ginkgo.By("Creating test NodePool to validate STS permissions")
+
+		refNodePool := existingNodePools.Items[0]
+		var subnet string
+		if spec, found, err := unstructured.NestedMap(refNodePool.Object, "spec"); found && err == nil {
+			if platform, found, err := unstructured.NestedMap(spec, "platform"); found && err == nil {
+				if aws, found, err := unstructured.NestedMap(platform, "aws"); found && err == nil {
+					if s, found, err := unstructured.NestedString(aws, "subnet"); found && err == nil {
+						subnet = s
+					}
+				}
+			}
+		}
```

Comment:
Would a helper function be more readable here?  You could give it a nodepool spec and a platform and it gives you a subnet?